packages/utils/src/consolidated/bulk-exists-consolidates.ts

```diff
@@ -0,0 +1,133 @@
+
+const cxId = getEnvVarOrFail("CX_ID");
+const medicalDocsBucketName = getEnvVarOrFail("MEDICAL_DOCUMENTS_BUCKET_NAME");
+const region = getEnvVarOrFail("AWS_REGION");
+const s3Utils = new S3Utils(region);
+
+// query stuff
+const confirmationTime = dayjs.duration(10, "seconds");
+
+// output stuff
+const getOutputFileName = buildGetDirPathInside(`exists-consolidated`);
+
+async function main() {
+  initRunsFolder();
+  const { log } = out("");
+
+  if (fileName) {
+    if (patientIds.length > 0) {
+      log(`>>> Patient IDs provided (${patientIds.length}), skipping file ${fileName}`);
+    } else {
+      const fileContents = getFileContents(fileName);
+      const idsFromFile = fileContents
+        .split(/\r?\n/)
+        .map(id => id.trim())
+        .filter(id => id.length > 0);
+      patientIds.push(...idsFromFile);
+      log(`>>> Found ${patientIds.length} patient IDs in ${fileName}`);
+    }
+  }
+
+  if (patientIds.length === 0) {
+    log(">>> No patient IDs provided. Please add patient IDs to the patientIds array.");
+    process.exit(1);
+  }
+
+  const startedAt = Date.now();
+  log(`>>> Starting at ${buildDayjs().toISOString()}with ${patientIds.length} patient IDs...`);
+
+  const { orgName } = await getCxData(cxId, undefined, false);
+  await displayWarningAndConfirmation(patientIds.length, orgName, log);
+
+  const existsFileName = getOutputFileName(orgName);
+  initFile(existsFileName);
+
+  log(`>>> Running it...`);
+  let ptIndex = 0;
+  await executeAsynchronously(
+    patientIds,
+    async patientId => {
+      await doesConsolidateExists(patientId, cxId, existsFileName, log);
+      log(`>>> Progress: ${++ptIndex}/${patientIds.length} patients complete`);
+    },
+    { numberOfParallelExecutions, minJitterMillis: 5, maxJitterMillis: 100 }
+  );
+  log(
+    `>>> Done checking consolidated data for ${patientIds.length} patients in ${elapsedTimeAsStr(
+      startedAt
+    )}`
+  );
+  process.exit(0);
+}
+
+async function displayWarningAndConfirmation(
+  patientCount: number | undefined,
+  orgName: string,
+  log: typeof console.log
+) {
+  const msg = `You are about to check if consolidated bundle exists for ${patientCount} patients of the org/cx ${orgName}.`;
+  log(msg);
+  log("Cancel this now if you're not sure.");
+  await sleep(confirmationTime.asMilliseconds());
+}
+
+async function doesConsolidateExists(
+  patientId: string,
+  cxId: string,
+  existsFileName: string,
+  log: typeof console.log
+) {
+  try {
+    const fileKey = createConsolidatedDataFilePath(cxId, patientId);
+    const exists = await s3Utils.fileExists(medicalDocsBucketName, fileKey);
+
+    if (exists) {
+      log(`">>> Consolidated data exists for patient ${patientId}...`);
+      fs.appendFileSync(existsFileName + "_exists.txt", `${patientId}\n`);
+    } else {
+      log(`    Consolidated data does not exist for patient ${patientId}... <<<<`);
+      fs.appendFileSync(existsFileName + "_missing.txt", `${patientId}\n`);
```

Comment:
This is a good idea. I need to do this with my SS scripts which generate CSV output - in general, would be good to see this as a "utils principle" that if the script stops running halfway through, there should at least be some record of the intermediate process.